File: OCDemo_v6.5.34/iOSDemo/Others/Widget/NormalNavBar.m
//
//  NormalNavBar.m
//  iOSDemo
//
//  Created by ltz on 2025/1/7.
//

#import "NormalNavBar.h"

@implementation NormalNavBar

- (instancetype)init
{
    self = [super init];
    if (self) {
        
        self.backgroundColor = UIColor.clearColor;
         
        self.bgImgView = [UIImageView new];
        self.bgImgView.backgroundColor = UIColor.redColor;
        self.bgImgView.image = kImg(@"BG");
        [self addSubview:self.bgImgView];
        
        [self.bgImgView mas_makeConstraints:^(MASConstraintMaker *make) {
            make.edges.equalTo(self);
        }];

        self.arrowImgView = [[WildClickButton alloc] init];
        [self.arrowImgView setImage:kImg(@"arrow_back") forState:0];
        [self addSubview:self.arrowImgView];
        
        [self.arrowImgView mas_makeConstraints:^(MASConstraintMaker *make) {
            make.bottom.mas_equalTo(self.mas_bottom).mas_offset(-kAdaptH(35, 35));
            make.width.mas_equalTo(25);
            make.height.mas_equalTo(25);
            make.left.mas_equalTo(self.mas_left).mas_offset(kAdaptW(32, 32));
        }];
        
        self.titleLbl = [UILabel new]; 
        self.titleLbl.textColor = kHexColor(0x1F2126);
        self.titleLbl.font = [UIFont boldSystemFontOfSize:18];
        self.titleLbl.textAlignment = NSTextAlignmentCenter;
        self.titleLbl.adjustsFontSizeToFitWidth = YES;
        [self addSubview:self.titleLbl];
        
        [self.titleLbl mas_makeConstraints:^(MASConstraintMaker *make) {
            make.centerY.mas_equalTo(self.arrowImgView.mas_centerY);
            make.centerX.mas_equalTo(self);
            make.height.mas_equalTo(self.titleLbl.font.lineHeight);
        }];
    }
    return self;
}

@end

File: SwiftDemo/SwiftDemo/Others/Widget/NormalNavBar.swift
//
//  NormalNavBar.swift
//  SwiftDemo
//
//  Created by ltz on 2025/8/20.
//

import UIKit
import SnapKit

class NormalNavBar: UIView {
    
    // MARK: - UI Properties
    var arrowImgView: WildClickButton!
    var bgImgView: UIImageView!
    var titleLbl: UILabel!
    
    // MARK: - Initializers
    override init(frame: CGRect) {
        super.init(frame: frame)
        setupUI()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setupUI()
    }
    
    // MARK: - Setup Methods
    private func setupUI() {
        backgroundColor = UIColor.clear
        
        // Background image view
        bgImgView = UIImageView()
        bgImgView.backgroundColor = UIColor.red
        bgImgView.image = kImg("BG")
        addSubview(bgImgView)
        
        bgImgView.snp.makeConstraints { make in
            make.edges.equalTo(self)
        }
        
        // Arrow button
        arrowImgView = WildClickButton()
        arrowImgView.setImage(kImg("arrow_back"), for: .normal)
        addSubview(arrowImgView)
        
        arrowImgView.snp.makeConstraints { make in
            make.bottom.equalTo(self.snp.bottom).offset(-kAdaptH(35, 35))
            make.width.equalTo(25)
            make.height.equalTo(25)
            make.left.equalTo(self.snp.left).offset(kAdaptW(32, 32))
        }
        
        // Title label
        titleLbl = UILabel()
        titleLbl.textColor = kHexColor(0x1F2126)
        titleLbl.font = UIFont.boldSystemFont(ofSize: 18)
        titleLbl.textAlignment = .center
        titleLbl.adjustsFontSizeToFitWidth = true
        addSubview(titleLbl)
        
        titleLbl.snp.makeConstraints { make in
            make.centerY.equalTo(arrowImgView.snp.centerY)
            make.centerX.equalTo(self)
            make.height.equalTo(titleLbl.font.lineHeight)
        }
    }
}
